parasoft-findings-teamcity-agent/src/test/java/com/parasoft/findings/teamcity/agent/ParasoftFindingsBuildProcessTest.java

```diff
@@ -0,0 +1,237 @@
+    }
+
+
+    private void testTransformStaticTestReport() throws Throwable {
+        File pmdReport = new File(reportDirPath + "/pmd-cpptest_professional_report.xml");
+        File pmdCpdReport = new File(reportDirPath + "/pmdCpd-cpptest_professional_report.xml");
+
+        try {
+            // Given
+            File testReport = new File(reportDirPath + "/cpptest_professional_report.xml");
+            params.put("reports.location", "reports/cpptest_professional_report.xml");
+            String pmdReportRelativePath = checkoutDir.toURI().relativize(pmdReport.toURI()).getPath();
+            prepareDataForTestTransformation();
+
+            doNothing().when(buildProgressLogger).message(Mockito.isA(String.class));
+            doNothing().when(buildProgressLogger).logMessage(Mockito.isA(BuildMessage1.class));
+
+            // When
+            BuildFinishedStatus status = parasoftFindingsBuildProcess.call();
+            String pmdReportSize = new DecimalFormat("0.00").format(pmdReport.length()/1024f);
+            int invalidReportCount = parasoftFindingsBuildProcess.getInvalidReportCount();
+
+            // Then
+            Assertions.assertTrue(pmdReport.exists());
+            Assertions.assertTrue(pmdCpdReport.exists());
+            Assertions.assertEquals(status.name(), "FINISHED_SUCCESS");
+            Assertions.assertEquals(invalidReportCount, 0);
+
+            // Verify the functions are called at least once
+            Mockito.verify(buildProgressLogger, Mockito.atLeastOnce()).message(Mockito.isA(String.class));
+            Mockito.verify(buildProgressLogger, Mockito.atLeastOnce()).logMessage(Mockito.isA(BuildMessage1.class));
+
+            // Verify the expected functions are called
+            Mockito.verify(buildProgressLogger).message("Transforming " + testReport.getAbsolutePath() + " with Parasoft Code Inspection");
+            Mockito.verify(buildProgressLogger).message("Generated report with transformation: " + pmdReport.getAbsolutePath());
+            Mockito.verify(buildProgressLogger).message("Importing data from '"+ pmdReportRelativePath + "' ("+ pmdReportSize +" KB) with 'message service' processor");
+            Mockito.verify(buildProgressLogger).message("Transforming " + testReport.getAbsolutePath() + " with Parasoft Duplicates");
+            Mockito.verify(buildProgressLogger).message("Generated report with transformation: " + pmdCpdReport.getAbsolutePath());
+        } finally {
+            clearGeneratedTestFile(pmdReport);
+            clearGeneratedTestFile(pmdCpdReport);
+        }
+    }
+
+    @Test
+    public void test_transformSOATestReport() throws Throwable {
+        File junitReport = new File(reportDirPath + "/junit-SOAtest_report.xml");
+        File pmdReport = new File( reportDirPath + "/pmd-SOAtest_report.xml");
+
+        // Given
+        try {
+            File testReport = new File(reportDirPath + "/SOAtest_report.xml");
```

Comment:
Please see if there's any room to refactor the process to read/process result and get build status. 
(For example, line 111 to line 120 here)